barretenberg/cpp/src/barretenberg/vm2/simulation/execution.cpp

```diff
@@ -172,6 +172,11 @@ void Execution::internal_return(ContextInterface& context)
     }
 }
 
+void Execution::keccak_permutation(ContextInterface& context, MemoryAddress dst_addr, MemoryAddress src_addr)
+{
+    keccakf1600.permutation(context, dst_addr, src_addr);
```

Comment:
should wrap this in a `try catch`. For now, just re-throw in the `catch` with a generic `std::runtime`, when @fcarreiro finishes his infra PR he'll make this an `OpCodeError` or something similar.